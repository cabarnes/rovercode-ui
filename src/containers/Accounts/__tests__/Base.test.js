import React from 'react';
import { Route } from 'react-router-dom';
import { shallow } from 'enzyme';

import Base from '../Base';

test('Base renders on the page with no errors', () => {
  const match = {
    path: '/accounts',
  };

  const wrapper = shallow(<Base match={match} />);

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find('img').exists()).toBe(true);
  expect(wrapper.find(Route).length).toBe(6);
  expect(wrapper.find(Route).at(0).prop('path')).toBe('/accounts/login');
  expect(wrapper.find(Route).at(1).prop('path')).toBe('/accounts/login/callback/:service');
  expect(wrapper.find(Route).at(2).prop('path')).toBe('/accounts/reset');
  expect(wrapper.find(Route).at(3).prop('path')).toBe('/accounts/reset/callback/:uid/:token');
  expect(wrapper.find(Route).at(4).prop('path')).toBe('/accounts/signup');
  expect(wrapper.find(Route).at(5).prop('path')).toBeUndefined();
});
