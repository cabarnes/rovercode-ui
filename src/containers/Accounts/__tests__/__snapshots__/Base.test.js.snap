// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Base renders on the page with no errors 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  container={true}
  direction="column"
  justify="center"
  spacing={3}
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <img
      alt="Rovercode"
      src={Object {}}
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Switch>
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "WrappedComponent": [Function],
            "displayName": "withCookies(injectIntl(Connect(Login)))",
            "render": [Function],
          }
        }
        exact={true}
        path="/accounts/login"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "WrappedComponent": [Function],
            "displayName": "withCookies(Connect(LoginCallback))",
            "render": [Function],
          }
        }
        exact={true}
        path="/accounts/login/callback/:service"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/accounts/reset"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/accounts/reset/callback/:uid/:token"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "WrappedComponent": [Function],
            "displayName": "withCookies(injectIntl(Connect(SignUp)))",
            "render": [Function],
          }
        }
        exact={true}
        path="/accounts/signup"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
